### PR TITLE
send metric for api response time

### PR DESCRIPTION
**Pull request by JosiasAurel**

### CODE REVIEW

**Comment by lachlanjc** [comment]
#### 📝 Changed routes:



- [**/api/metric**](https://scrapbook-7ntck078a.hackclub.dev/api/metric)



Commit b7cd24be533f3014aa18f06452bebc1adbe21f2d (https://scrapbook-7ntck078a.hackclub.dev).

File: middleware.js [hunk 1]
```diff
@@ -44,20 +46,23 @@ export async function middleware(req) {
...
       });
+      time = (new Date().getTime()) - startTime;
     }
 
     const data = await response.json();
 
     // attempt to send metric
     // will timeout after 150ms
-    Promise.any([createTimeoutPromise(150), sendMetric(HOST_NAME, `${response.status}.${_metricName}`)])
+    Promise.any([createTimeoutPromise(150), sendMetric(HOST_NAME, `${response.status}.${_metricName}`, time)])
```
Comment (grymmy) [inline middleware.js:65]: I believe this should instead be a tuple of three promises, not two - one timeout, time for the timer, one for the counter

**Review by grymmy — CHANGES_REQUESTED** [review]
The api/metrics endpoint does not allow for passing through a 'type' for the metrics - this solution is workable (exposing an endpoint that the middleware hits for sending the metrics), but we need to support all metric types at least...

**Review by grymmy — APPROVED** [review]
Upon further thought, this is okay as long as we do not intend on using api/metric scrapbook endpoint for anything other than jailbreaking edge runtime in middleware.

**Pull request merged**